graphql-test/src/main/java/org/jahia/test/graphql/GraphQLPublicationTest.java

```diff
@@ -296,4 +296,22 @@ private JCRSiteNode createTestSite(String name, String template, Set<String> lan
         defaultSession.save();
         return site;
     }
+
+    private void waitForPublicationToFinish() {
+        SchedulerService schedulerService = BundleUtils.getOsgiService(SchedulerService.class, null);
+        try {
+            with().pollInterval(ONE_SECOND).await().atMost(TIMEOUT_WAITING_FOR_PUBLICATION, MILLISECONDS).until(new Callable<Boolean>() {
+                @Override public Boolean call() throws Exception {
+                    final String[] triggerNames = schedulerService.getScheduler().getTriggerNames(SchedulerService.INSTANT_TRIGGER_GROUP);
+                    final List<JobDetail> allActiveJobs = schedulerService.getAllActiveJobs();
+                    final boolean publicationJobs = allActiveJobs.stream().anyMatch(job -> job.getDescription().equals("Publication"));
+                    logger.debug("Trigger names: {}", Arrays.toString(triggerNames));
+                    allActiveJobs.forEach(j -> logger.debug("Active job '{}'", j.getDescription()));
+                    return Arrays.stream(triggerNames).anyMatch(triggerName -> triggerName.contains("Publication")) && publicationJobs;
+                }
+            }, equalTo(false));
+        } catch (ConditionTimeoutException e) {
+            Assert.fail("Publication job did not finish within the set timeout of " + TIMEOUT_WAITING_FOR_PUBLICATION + " ms");
+        }
+    }
```

Comment:
For me the method is difficult to read and understand what it is doing. Also it is not necessary to check for triggers and active jobs in each poll iteration. I would rather suggest something like that:

```
    private void waitForPublicationToFinish() {
        SchedulerService schedulerService = BundleUtils.getOsgiService(SchedulerService.class, null);
        ConditionFactory conditionalWait = with().pollInterval(ONE_SECOND).await().atMost(TIMEOUT_WAITING_FOR_PUBLICATION, MILLISECONDS);
        // wait until no more PublicationJob triggers are scheduled 
        conditionalWait.until(() -> Arrays.stream(schedulerService.getScheduler().getTriggerNames(SchedulerService.INSTANT_TRIGGER_GROUP))
                .noneMatch(triggerName -> triggerName.contains(PUBLICATION_JOB_NAME)));
        // wait until last PublicationJob is finished
        conditionalWait.until(() -> schedulerService.getAllActiveJobs().stream()
                .noneMatch(job -> job.getJobClass().getName().contains(PUBLICATION_JOB_NAME)));
    }
```

The added logging and the exeption catch/assertion IMHO are not necessary.